File: leetcode/pattern/1-prefix-sum/1-prefix-sum.py
class Solution:
  def prefix_sum(arr, start, end):
    new_arr = arr[start:end+1]
    
    sum = 0
    for x in new_arr:
      sum += x

    return sum
  

s = Solution
print(
  9 == s.prefix_sum([1, 2, 3, 4, 5, 6], 1, 3),
  21 == s.prefix_sum([1, 2, 3, 4, 5, 6], 0, 5)
)

File: leetcode/pattern/1-prefix-sum/1-prefix-sum.go
package main

import (
	"fmt"
	"reflect"
)

func main() {
	got := prefixSum([]int{1, 2, 3, 4, 5, 6}, 1, 3)
	want := 9
	if !reflect.DeepEqual(got, want) {
		fmt.Println("test failed")
		fmt.Printf("expected %d, but got %d\n", want, got)
	} else {
		fmt.Println("test passed")
	}
}

func prefixSum(nums []int, start int, end int) int {
	newArr := nums[start : end+1]
	result := 0

	for _, v := range newArr {
		result += v
	}

	return result
}
